### PR TITLE
Ui refactorings, minor bugfixes and polish

### PR DESCRIPTION
**Pull request by toolness**
Here are a bunch of UI refactorings, and a few minor bugfixes and UI polish improvements.

### CODE REVIEW

**Comment by toolness** [comment]
All QUnit tests pass on Firefox, IE9, and Chrome.

File: js/fc/ui/modals.js [hunk 1]
```diff
@@ -28,73 +26,139 @@ define(function() {
...
     
-    $("#publish-button").click(function(){
-      if ($(this).hasClass("enabled")) $("#confirm-dialog").fadeIn();
+    publishButton.click(function(){
+      if ($(this).hasClass("enabled")) confirmDialog.fadeIn();
     });
+    
+    $("#confirm-publication", confirmDialog).click(function(){
+      // Reset the publish modal.
+      shareResult.unbind('.hotLoad');
```
Comment (Pomax) [inline js/fc/ui/modals.js:93]: should this be unbind or removeClass?

File: js/fc/ui/social-media.js [hunk 2]
```diff
@@ -20,7 +17,15 @@ define(function() {
       facebook: {
         id: "facebook-jssdk",
         src: "//connect.facebook.net/en_US/all.js#xfbml=1&appId=1",
-        html: "<div class='fb-like' data-href='"+urlPlaceHolder+"' data-send='false' data-action='recommend' data-layout='button_count' data-show-faces='false' data-font='tahoma'></div>"
+        html: "<div class='fb-like' data-href='"+urlPlaceHolder+"' data-send='false' data-action='recommend' data-layout='button_count' data-show-faces='false' data-font='tahoma'></div>",
+        afterHotLoad: function() {
+          // Facebook needs additional help, because it needs
+          // to be told that it has to refresh its button, rather
+          // than simply reloading.
+          if (typeof(FB) == "object" && FB.XFBML && FB.XFBML.parse) {
```
Comment (Pomax) [inline js/fc/ui/social-media.js]: === rather than ==?

File: js/main.js [hunk 3]
```diff
@@ -261,7 +217,15 @@ define("main", function(require) {
       doneLoading();
     }, "text");
   } else
-    publishUI.loadCode(pageToLoad, doneLoading);
+    publisher.loadCode(pageToLoad, function(err, data, url) {
+      if (err) {
+        alert('Sorry, an error occurred while trying to get the page.');
```
Comment (Pomax) [inline js/main.js:222]: should this get [...] "the page: " + err?

File: js/fc/ui/modals.js (hunk 1 above)
Comment (toolness) [inline js/fc/ui/modals.js:93]: `unbind()` for sure... see the section on **Using Namespaces** in the [unbind documentation](http://api.jquery.com/unbind/).

File: js/fc/ui/social-media.js (hunk 2 above)
Comment (toolness) [inline js/fc/ui/social-media.js]: Fixed in 9198eee0bc50dca8d283ea29a7a1322e27a59dfe !

File: js/main.js (hunk 3 above)
Comment (toolness) [inline js/main.js:222]: I think this should probably just be filed as a separate bug for better error reporting, as fixing this is a bit non-trivial--jQuery's ajax error reporting callback returns different information in different arguments that we'd want to do a bit of logic with to provide users with reasonably useful information.

Comment (toolness) [inline js/main.js:222]: Filed as #113.

**Pull request merged**